codec/std/codec.go

```diff
@@ -117,24 +121,22 @@ func (c *Codec) UnmarshalSupplyJSON(bz []byte) (bankexported.SupplyI, error) {
 // the Marshaler interface, it is treated as a Proto-defined message and
 // serialized that way. Otherwise, it falls back on the internal Amino codec.
 func (c *Codec) MarshalEvidence(evidenceI eviexported.Evidence) ([]byte, error) {
```

Comment:
The only reasons would be if a) people wanted to use amino-encoding for state or b) people wanted to optimize state encoding with `oneof`s. The main reason for allowing amino before was I believe because our protobuf approach required app-level `oneof`s whereas with amino it was much more flexible. With the `Any` approach that argument doesn't really hold anymore. So it's really just if we want to allow someone to optimize space... What do you think?

Reply:
I see. I'm suggesting we still keep the codec interfaces in the respective modules. However, I'm proposing we remove the notion and implementation of an app-level codec type/implementation.

Applications that want `oneofs` will just implement their own app-level codec. I still see a strong argument to keep it.

Reply:
Okay, so just have `EvidenceCodec` as a type in `x/evidence` and that takes `Marshaler` as an argument?

Reply:
Yup :)

Reply:
Okay I'll refactor

Reply:
It should be minimal since it can be done piecemeal. Just remove the evidence related methods from the app-codec.